consular/tests/test_main.py

```diff
@@ -1110,6 +1111,95 @@ def test_purge_dead_services(self):
         yield d
 
     @inlineCallbacks
+    def test_purge_task_lost_services(self):
+        d = self.consular.purge_dead_services()
+        consul_request = yield self.requests.get()
+        consul_request['deferred'].callback(
+            FakeResponse(200, [], json.dumps([{
+                'Node': 'consul-node',
+                'Address': '1.2.3.4',
+            }]))
+        )
+
+        agent_request = yield self.requests.get()
+        # Expecting a request to list of all services in Consul,
+        # returning 2
+        self.assertEqual(
+            agent_request['url'],
+            'http://1.2.3.4:8500/v1/agent/services')
+        self.assertEqual(agent_request['method'], 'GET')
+        agent_request['deferred'].callback(
+            FakeResponse(200, [], json.dumps({
+                "testinggroup-someid1": {
+                    "ID": "taskid1",
+                    "Service": "testingapp",
+                    "Tags": None,
+                    "Address": "machine-1",
+                    "Port": 8102,
+                    "Tags": [
+                        "consular-reg-id=test",
+                        "consular-app-id=/testinggroup/someid1",
+                    ],
+                },
+                "testinggroup-someid1": {
+                    "ID": "taskid2",
+                    "Service": "testingapp",
+                    "Tags": None,
+                    "Address": "machine-2",
+                    "Port": 8103,
+                    "Tags": [
+                        "consular-reg-id=test",
+                        "consular-app-id=/testinggroup/someid1",
+                    ],
+                }
+            }))
+        )
+
+        # Expecting a request for the tasks for a given app, returning
+        # 1 of which only has the TASK_RUNNING state.
```

Comment:
This comment doesn't make much sense. Maybe "returning 2 of which only 1 has the TASK_RUNNING state"?